core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala

```diff
@@ -87,19 +110,31 @@ class ExecutorsListener(storageStatusListener: StorageStatusListener, conf: Spar
         s.blockManagerId.executorId == SparkContext.LEGACY_DRIVER_IDENTIFIER ||
         s.blockManagerId.executorId == SparkContext.DRIVER_IDENTIFIER
       }
-      storageStatus.foreach { s => executorToLogUrls(s.blockManagerId.executorId) = logs.toMap }
+      storageStatus.foreach { s =>
+        val eid = s.blockManagerId.executorId
+        if (!executorToTaskSummary.contains(eid)) {
```

Comment:
Hm, and while we're at it, maybe these blocks that check if a key has a mapping and otherwise add a new value can use `getOrElseUpdate`? it might only work if `ExecutorTaskSummary` does not need the "executorId" field, but does it need that? it's kind of redundant since it's always looked up by key. Not sure, maybe I missed something.